js/core/structure.js

```diff
@@ -70,25 +70,25 @@ define(
 
         var $a = $("<a/>").attr({ href: "#" + id, "class": "tocxref" })
           .append(isIntro ? "" : $span.clone())
-          .append($kidsHolder.contents()),
-          $item = $("<li class='tocline'/>").append($a);
-        if (conf.maxTocLevel === 0 || level <= conf.maxTocLevel) $ul.append($item);
+          .append($kidsHolder.contents());
+        var $item = $("<li class='tocline'/>").append($a);
+        if (conf.maxTocLevel === 0 || level <= conf.maxTocLevel) $ol.append($item);
         current.push(0);
         var $sub = makeTOCAtLevel($sec, doc, current, level + 1, conf);
         if ($sub) {
           $item.append($sub);
         }
         current.pop();
       }
-      return $ul;
+      return $ol;
     }
 
     return {
       run: function(conf, doc, cb) {
-        if (!conf.tocIntroductory) {
+        if ("tocIntroductory" in conf === false) {
```

Comment:
The problem with the old check is that it's checking for a falsy value, but then overwriting it irrespective of it being there (that's a bug - because if the value is there as null, 0, false, or undefined, it should be left alone). 

What the code needs to be checking is checking if the property was defined, and if not, setting it to the default of false (new code).


> On 5 Aug 2016, at 2:02 AM, Shane McCarron <notifications@github.com> wrote:
> 
> In js/core/structure.js:
> 
> >      }
> >  
> >      return {
> >        run: function(conf, doc, cb) {
> > -        if (!conf.tocIntroductory) {
> > +        if ("tocIntroductory" in conf === false) {
> I dont mind, but I am curious why you prefer this idiom? It seems inconsistent with the rest of the code.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>